ietf/templates/meeting/agenda.html

```diff
@@ -154,35 +154,27 @@ <h2>
                     </span>
                       </td>
                       <td colspan="3">
-                    <span class="hidden-sm hidden-md hidden-lg">
-                       {% include "meeting/timeslot_start_end.html" %}
-                    </span>
-                          {% if item.timeslot.show_location and item.timeslot.get_html_location %}
-                              {% if schedule.meeting.number|add:"0" < 96 %}
-                                  <a href="https://tools.ietf.org/agenda/{{schedule.meeting.number}}/venue/?room={{ item.timeslot.get_html_location|xslugify }}">{{item.timeslot.get_html_location}}</a>
-                              {% elif item.timeslot.location.floorplan %}
-                                  <a href="{% url 'ietf.meeting.views.floor_plan' num=schedule.meeting.number %}?room={{ item.timeslot.get_html_location|xslugify }}">{{item.timeslot.get_html_location}}</a>
-                              {% else %}
-                                  {{item.timeslot.get_html_location}}
-                              {% endif %}
-                              {% with item.timeslot.location.floorplan as floor %}
-                                  {% if item.timeslot.location.floorplan %}
-                                      <span class="hidden-xs">
-			<a href="{% url 'ietf.meeting.views.floor_plan' num=schedule.meeting.number %}#{{floor.name|xslugify}}"
-               class="pull-right" title="{{floor.name}}"><span class="label label-blank label-wide">{{floor.short}}</span></a>
-                      </span>
-                                  {% endif %}
-                              {% endwith %}
-                          {% endif %}
+                        <span class="hidden-sm hidden-md hidden-lg">
+                           {% include "meeting/timeslot_start_end.html" %}
+                        </span>
+                        {% location_anchor item.timeslot %}
+                          {{ item.timeslot.get_html_location }}
+                        {% end_location_anchor %}
+                        {% if item.timeslot.show_location and item.timeslot.get_html_location %}
```

Comment:
Quite an improvement.

Reply:
:+1: That, and the fact that it happens 3 times on the page, was my motivation for this slightly off-topic project.

(I did start to do away with that repetition, but it gets hairier than seemed appropriate for a side track)